back-nest/src/products/products.contoller.ts

```diff
@@ -11,4 +11,10 @@ export class ProductController {
   async getAllProducts(): Promise<product[]> {
     return this.productService.getAllProducts();
   }
+
+  @Delete(':id')
+  async deleteProduct(@Param('id') id: string) {
+    const productId = +id;
+    return this.productService.deleteProduct(productId);
```

Comment:
There's no need to return that method

```suggestion
    const productId = +id;
    await this.productService.deleteProduct(productId);
```